### ISSUE TITLE
(node:5580) UnhandledPromiseRejectionWarning: TimeoutError: Navigation Timeout Exceeded: 30000ms exceeded

### ISSUE THREAD

**Issue by FraH90** [snobu/destreamer#18]
So, it was working for me till I updated the script with your latest version; i updated because sometimes it worked, other times it didn't (very strange) so I tried to update; I did npm install, npm run -s build, and added --no-sandbox to arguments, just like I did the first time I installed it. 

Now, it gives me this:
`
Launching headless Chrome to perform the OpenID Connect dance...
Navigating to STS login page...
We are logged in. Sorry, i mean "you".
(node:5580) UnhandledPromiseRejectionWarning: TimeoutError: Navigation Timeout Exceeded: 30000ms exceeded
    at C:\Users\xxx\Desktop\destreamer-dev\node_modules\puppeteer\lib\LifecycleWatcher.js:142:21
  -- ASYNC --
    at Frame.<anonymous> (C:\Users\xxx\Desktop\destreamer-dev\node_modules\puppeteer\lib\helper.js:111:15)
    at Page.goto (C:\Users\xxx\Desktop\destreamer-dev\node_modules\puppeteer\lib\Page.js:674:49)
    at Page.<anonymous> (C:\Users\xxx\Desktop\destreamer-dev\node_modules\puppeteer\lib\helper.js:112:23)
    at rentVideoForLater (C:\Users\xxx\Desktop\destreamer-dev\destreamer.js:72:20)
(node:5580) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:5580) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`

I've also tried to reinstall everything, with the old version of the script (I had it zipped), but nothing ;(
Please help me, I'm using your script to download the videos of my class :(

PS if you need the logs you can tell me where to find them (I'm not pratical of nodejs)

**Comment by snobu** [comment]
I can't seem to reproduce this on either Node 8 / 10 / 12 or 13.
Does the browser window navigate to the video URL after logging in?

**Comment by klenium** [comment]
Same happened for me. A new video was just recorded in a MS Teams meeting. After it finished, the video was available in Stream to play in real browser, but the script failed to download it. I waited like one hour, tried it again and I could download it. Maybe in the background there is some conversion?

**Comment by klenium** [comment]
> Does the browser window navigate to the video URL after logging in?
Yes. In this test-managed-browser, we can't watch the video, because this type of browser is not supported by Stream. However, in other videos I could see the actual video for about 0.1 seconds. In the case the error occurred, I didn't see any part of the video.

**Comment by snobu** [comment]
Can you navigate to this URL

```
https://use2-2.api.microsoftstream.com/api/videos/THE_GUID_OF_YOUR_VIDEO/?api-version=1.3-private
```

and check if `playbackUrls` being returned in the JSON response has an entry with this mime type?

```
"mimeType": "application/vnd.apple.mpegurl"
```

**Comment by FraH90** [comment]
> > Does the browser window navigate to the video URL after logging in?
> 
> Yes. In this test-managed-browser, we can't watch the video, because this type of browser is not supported by Stream. However, in other videos I could see the actual video for about 0.1 seconds. In the case the error occurred, I didn't see any part of the video.

I was saying the same exact thing. When the software was working right, you could watch a preview of the video, for 1 second; but when you have that error, the preview doesn't show.
Don't know if it can help knowing this behaviour.

PS: I've tried many and many times, without doing things differently... the last time for me it worked!
But I swear, I've not done anything differently, and it was the same link

**Comment by klenium** [comment]
> ```
> https://use2-2.api.microsoftstream.com/api/videos/THE_GUID_OF_YOUR_VIDEO/?api-version=1.3-private
> ```
That returns a 401 error.

**Comment by Dam98** [comment]
> Can you navigate to this URL
> 
> ```
> https://use2-2.api.microsoftstream.com/api/videos/THE_GUID_OF_YOUR_VIDEO/?api-version=1.3-private
> ```
> 
> and check if `playbackUrls` being returned in the JSON response has an entry with this mime type?
> 
> ```
> "mimeType": "application/vnd.apple.mpegurl"
> ```
> 
> Login to Microsoft Stream first in your browser of choice.

{
  "error": {
    "code": "InvalidUrlId",
    "message": "The URL resource id is invalid."
  }
}

**Comment by klenium** [comment]
@Dam98 You have to replace `THE_GUID_OF_YOUR_VIDEO` in that URL to the real value.

**Comment by FraH90** [comment]
Same for me. For the videos that I'm not able to download it gives a 404 error

**Comment by FraH90** [comment]
> I can't seem to reproduce this on either Node 8 / 10 / 12 or 13.
> Does the browser window navigate to the video URL after logging in?

Ok, I have news on the bug: I can confirm that it is totally random. 
It doesn't happen always now, but just with some videos: the strange thing is that after I've get the error, if I try again with the same video another time, it will work!
But it's annoying, cause you need to login everytime, and moreover sometimes you need to try two or three more times to make it work.

However, thanks for your beautiful work, I really appreciate it and hope it will be improved by the community.

**Comment by snobu** [comment]
We're changing the way we fetch the Media Services video URL in [this PR](https://github.com/snobu/destreamer/pull/20). There's real hope things are going to get way more stable right after.

**Comment by lukaarma** [comment]
This issue seems almost resolved on my fork thanks to a change in the way puppeteer waits for the video page to load, we'll see if it's resolved for everyone after #20 will be merged in the repo

**Comment by snobu** [comment]
This should be fixed now. Just reopen if you still see this problem.
